src/Microsoft.Framework.Logging/LoggerStructureFormat.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+
+namespace Microsoft.Framework.Logging
+{
+    public class LoggerStructureFormat : ILoggerStructure
+    {
+        static Dictionary<string, Formatter> _formatters = new Dictionary<string, Formatter>();
+        static object _formattersLock = new Object();
+        private readonly Formatter _formatter;
+        private readonly object[] _values;
+
+        public LoggerStructureFormat(string format, params object[] values)
+        {
+            lock (_formattersLock)
+            {
+                // TODO: ConcurrentDictionary not available in portable profile?
+                // _formatter = _formatters.GetOrAdd(format, ParseFormatter);
+                if (!_formatters.TryGetValue(format, out _formatter))
+                {
+                    _formatter = ParseFormatter(format);
+                    _formatters[format] = _formatter;
+                }
+            }
+            _values = values;
+        }
+
+        public string Message { get { return null; } }
+
+        public string Format()
+        {
+            return _formatter.Format(_values);
+        }
+
+        public IEnumerable<KeyValuePair<string, object>> GetValues()
+        {
+            return _formatter.GetValues(_values);
+        }
+
+        private Formatter ParseFormatter(string format)
+        {
+            return new Formatter(format);
+        }
+
+        class Formatter
+        {
+            private readonly string _format;
+            private readonly List<string> _valueNames = new List<string>();
+
+            public Formatter(string format)
+            {
+                var sb = new StringBuilder();
+                var endIndex = format.Length;
+                for (var scanIndex = 0; scanIndex != endIndex;)
+                {
+                    var openBraceIndex = FindIndexOf(format, '{', scanIndex, endIndex);
+                    var closeBraceIndex = FindIndexOf(format, '}', openBraceIndex, endIndex);
+
+                    // Format item syntax : { index[,alignment][ :formatString] }.
```

Comment:
@lodejard I had to update this as the format item syntax is per the comment here.